File: Mark/Lists/Day-5/Plus One/plus-one.py
class Solution:
    def plusOne(self, digits: list[int]) -> list[int]:
        num = []
        for digit in digits:
            num.append(str(digit))
        result = int(''.join(num))
        result = result + 1
        result_to_str = str(result)
        num.clear()
        
        for i in result_to_str:
            num.append(i)
        return num

        

solution = Solution()
output = solution.plusOne([9])
print(output)

File: Olix/Arrays/day-5/plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function (digits) {
  // let lastIndexPosition = [digits.length - 1];
  // if (digits[lastIndexPosition] < 9) {
  //   digits[lastIndexPosition] = digits[lastIndexPosition] + 1;
  // } else if (digits[lastIndexPosition] >= 9) {
  //   let digit = BigInt(digits.join("")) + 1n;
  //   return String(digit)
  //     .split("")
  //     .map((num) => Number(num));
  //   // return BigInt(digits.join("")) + 1n;
  return ("" + (BigInt(digits.join("")) + 1n)).split("");
  // const num = "" + (BigInt(digits.join("")) + 1n);
};

console.log(plusOne([2, 3]));
// BigInt(digits.join("")) + 1n;
